Create Auto-Drive Treasury Contracts (Buy Credits PR #1) (#485)
autonomys/auto-drive: packages/contracts/README.md

## AutoDriveCreditsReceiver

**Purpose**: Minimal ETH receiver that tags incoming deposits with an `intentId` and later sweeps funds to a designated treasury. Owner can pause operations and update the treasury using two-step ownership.

### Inheritance

- `Ownable2Step`
- `ReentrancyGuard`
- `Pausable`

### State

- `address payable public treasury` — destination for swept funds. Must be set by the owner before sweeping.

### Events

> - `Deposit(bytes32 indexed intentId, uint256 depositAmount)` — emitted on each deposit with the provided intent identifier.

Reviewer: ```suggestion
- `IntentPaymentReceived(bytes32 indexed intentId, uint256 depositAmount)` — emitted on each deposit with the provided intent identifier.
```

- `TreasuryUpdated(address indexed oldTreasury, address indexed newTreasury)` — emitted when the owner updates the treasury.
- `SweptToTreasury(address indexed caller, uint256 amount)` — emitted when funds are swept to the treasury.

### Functions

> - `deposit(bytes32 intentId) public payable whenNotPaused`
>   - Accepts ETH and emits `Deposit` with the `intentId` and `msg.value`.

Reviewer: ```suggestion
- `payIntent(bytes32 intentId) public payable whenNotPaused`
  - Accepts ETH and emits `IntentPaymentReceived` with the `intentId` and `msg.value`.
```

  - No minimum deposit enforced; caller supplies the payable value.

- `setTreasury(address payable newTreasury) public onlyOwner`
  - Sets the treasury address. Reverts if `newTreasury == address(0)`.
  - Emits `TreasuryUpdated(old, newTreasury)`.

- `sweepAmountToTreasury(uint256 amount) public nonReentrant whenNotPaused`
  - Transfers exactly `amount` of the contract's ETH balance to `treasury`.
  - Reverts if treasury unset, amount is zero, or insufficient balance.

>   - Emits `SweptToTreasury(msg.sender, amount)`.

Reviewer: ```suggestion
  - Emits `SweptToTreasury(msg.sender, treasury, amount)`.
```

- `sweepAllToTreasury() public nonReentrant whenNotPaused`
  - Transfers the entire ETH balance to `treasury`.
  - Reverts if treasury unset.

>   - Emits `SweptToTreasury(msg.sender, balance)`.

Reviewer: ```suggestion
  - Emits `SweptToTreasury(msg.sender, treasury, balance)`.
```
